frontend/src/components/admin/menu/DictionaryManagement.js

```diff
@@ -0,0 +1,349 @@
+  const [dictionaryEntry, setDictionaryEntry] = useState("");
+  const [localAbbreviation, setLocalAbbreviation] = useState("");
+  const [isActive, setIsActive] = useState("");
+
+  const handlePageChange = (pageInfo) => {
+    if (page != pageInfo.page) {
+      setPage(pageInfo.page);
+    }
+
+    if (pageSize != pageInfo.pageSize) {
+      setPageSize(pageInfo.pageSize);
+    }
+  };
+
+  const fetchedDictionaryMenu = (dictionaryMenus) => {
+    if (componentMounted.current) {
+      setDictionaryMenuz(dictionaryMenus);
+    }
+  };
+
+  const fetchedDictionaryCategory = (category) => {
+    if (componentMounted.current) {
+      console.log("Fetched Dictionary Category Data:", category);
+      setCategoryDescription(category);
+    }
+  };
+
+  useEffect(() => {
+    componentMounted.current = true;
+    getFromOpenElisServer("/rest/get-dictionary-menu", fetchedDictionaryMenu);
+    return () => {
+      componentMounted.current = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    componentMounted.current = true;
+    getFromOpenElisServer(
+      "/rest/dictionary-categories/descriptions",
+      fetchedDictionaryCategory
+    );
+    return () => {
+      componentMounted.current = false;
+    };
+  }, []);
+
+  const postData = {
+    id: dictionaryNumber,
+    selectedDictionaryCategoryId: category,
+    dictEntry: dictionaryEntry,
+    localAbbreviation: localAbbreviation,
+    isActive: isActive,
+  };
+
+  async function displayStatus(res) {
+    setNotificationVisible(true);
+    if (res.status == "201") {
+      addNotification({
+        kind: NotificationKinds.success,
+        title: intl.formatMessage({ id: "notification.title" }),
+        message: intl.formatMessage({ id: "success.add.edited.msg" }),
+      });
+    } else {
+      addNotification({
+        kind: NotificationKinds.error,
+        title: intl.formatMessage({ id: "notification.title" }),
+        message: intl.formatMessage({ id: "error.add.edited.msg" }),
+      });
+    }
+    reloadConfiguration();
+  }
+
+  const handleSubmitModal = (e) => {
+    e.preventDefault();
+    console.log(JSON.stringify(postData));
+    postToOpenElisServerFullResponse(
+      "/rest/create-dictionary",
+      JSON.stringify(postData),
+      displayStatus
+    );
+    setOpen(false);
+  };
+
+  return (
+    <>
+      {notificationVisible === true ? <AlertDialog /> : ""}
+      <PageBreadCrumb
+        breadcrumbs={[
+          { label: "home.label", link: "/" },
+          { label: "master.lists.page", link: "/MasterListsPage" },
+          { label: "dictionary.label.modify", link: "/DictionaryManagement" },
+        ]}
+      />
+
+      <Grid fullWidth={true}>
+        <Column lg={16}>
+          <Section>
+            <Section>
+              <Heading>
+                <FormattedMessage id="dictionary.label.modify" />
+              </Heading>
+            </Section>
+          </Section>
+        </Column>
+      </Grid>
+
+      <div className="orderLegendBody">
```

Comment:
We want this to display with in  the admin page , can u use the adminPageContent class ie [here ](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/develop/frontend/src/components/admin/labNumber/LabNumberManagement.js#L160)